### ISSUE TITLE
providing extended project analysis

### ISSUE THREAD

**Issue by hamidzr** [NetsBlox/NetsBlox#1881]
I'm thinking about having an endpoint on the server which provides basic analysis on projects. This information can then be used in different views and interfaces that we have to first give the users more information and secondly help us build smarter UIs (like in case of the mobile application)
- information about roles
- services
- sensor usage
etc

**Comment by brollb** [comment]
I think this would be better to implement as a standalone service and use a public API defined by netsblox. We might already have enough of a public API, actually. For example, we could have a new analytics server which queries public projects and runs the analytics on them. For private projects, we could have the users log in, of course.

What types of analytics are you thinking?

**Comment by hamidzr** [comment]
It came up because we are starting to support different environment w/ diff requirements and it would be good for example if we could see that the project requires network access (does it have multiple roles? does it call any RPCs? ) or if the project is using mobile-specific sensors or is connecting to a robot etc.

**Comment by brollb** [comment]
Makes sense. Those analytics would probably be good to have. It also might be interesting to get information about the total number of events in each project (to give an idea of relative size), number of collaborative projects, and total duration that the project has been worked on over.

I think the most pressing question is if these analytics should be included in netsblox itself or another server (it could be hosted at something like `analytics.netsblox.org`). My vote would be to make another project and server for these capabilities to keep the codebases smaller and more manageable. Additionally, this would allow us to minimize the load of the main netsblox server (which we definitely don't want to add bottlenecks to).
